Sources/GRPCNIOTransportCore/Client/Resolver/NameResolver+IPv6.swift

```diff
@@ -29,6 +30,23 @@ extension ResolvableTargets {
     /// Create a new IPv6 target.
     /// - Parameter addresses: The IPv6 addresses.
     public init(addresses: [SocketAddress.IPv6]) {
+      debugOnly {
```

Comment:
Wouldn't it also be useful to get some better error other than [the current](https://github.com/grpc/grpc-swift-nio-transport/issues/116#issue-3223783756) `unavailable: "Channel isn't ready."` in release builds, too? 

I'm thinking if the address comes from some config and there are different values for debug and release, it's possible that debug builds won't fail but then you'd get some confusing error in release.

Reply:
It would be useful but at that layer you don't have enough information to tie it back to creating the wrong resolver. The "Channel isn't ready." message can happen for any number of reasons, including you could've correctly used a resolver but the server just isn't up yet.

Reply:
I guess my question was actually if we should _always_ perform this check, not just in debug mode (throwing an error instead of failing an assertion), or if it's too expensive to afford it in this codepath.

Reply:
I don't think it's necessary tbh, the change in spelling is clear enough.